### DISCUSSION TITLE
Call iotube contracts from ioctl CLI?

### DISCUSSION THREAD

**Discussion by anuade** [iotexproject/ioTube#70] [General]
Hi,

Well done on this work. 
Would it be possible to extend this to third party apps which wrap the ioctl CLI? The idea is to be able to perform cross-chain conversions leveraging the iotube contracts. I am building a similar app at the moment and would be interested to know if this possible. Thanks.

**Comment by guo** [comment]
Sure you can. you can use regular web3 function calls to do that. just use ABI's to call.

**Comment by anuade** [comment]
Thanks. I'm looking into that. Just wondering: do I need to make the one function call at the entry point/token of the tube and it all gets sorted from there it do I have to make the calls to each contract involved from start to finish?

________________________________
